src/main/java/org/bihealth/mi/easybus/implementations/email/BusEmail.java

```diff
@@ -139,42 +164,44 @@ public void stop() {
     
     /**
      * Receives e-mails
-     * @throws BusException 
      * @throws InterruptedException 
      */
-    private synchronized void receiveEmails() throws BusException, InterruptedException {
-        
-        // Get mails
-        BusEmail.BusEmailMessage deleted = null;
-        for (BusEmail.BusEmailMessage message : connection.receive()) {
+    private synchronized void receiveEmails() throws InterruptedException {
+        try {
+            // Get mails
+            BusEmail.BusEmailMessage deleted = null;
+            for (BusEmail.BusEmailMessage message : connection.receive()) {
 
-            // Check for interrupt
-            if (Thread.interrupted()) {
-                connection.close();
-                throw new InterruptedException();
-            }
+                // Check for interrupt
+                if (Thread.interrupted()) {
+                    connection.close();
+                    throw new InterruptedException();
+                }
 
-            // Mark
-            boolean received = false;
-            
-            // Send to scope and participant
-            try {
-                received |= receiveInternal(message.message, message.scope, message.receiver);
-            } catch (InterruptedException e) {
-                connection.close();
-                throw e;
+                // Mark
+                boolean received = false;
+
+                // Send to scope and participant
+                try {
+                    received |= receiveInternal(message.message, message.scope, message.receiver);
+                } catch (InterruptedException e) {
+                    connection.close();
+                    throw e;
+                }
+
+                // Delete
+                if (received) {
+                    message.delete();
+                    deleted = message;
+                }
             }
 
-            // Delete 
-            if (received) {
-                message.delete();
-                deleted = message;
+            // Expunge
+            if (deleted != null) {
+                deleted.expunge();
             }
-        }
-        
-        // Expunge
-        if (deleted != null) {
-            deleted.expunge();
+        } catch (BusException e) {
+            // Ignore
```

Comment:
I'm not convinced that it is a good idea to no throw/catch BusException anymore. Please reconsider and/or explain.

Reply:
See first answered comment